src/components/PostDetails.tsx

```diff
@@ -154,18 +106,20 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
           </>
         )}
 
-        {loaded && !hasError && !visible && (
+        {!loaded && !hasError && !visible && (
           <button
             data-cy="WriteCommentButton"
             type="button"
             className="button is-link"
-            onClick={() => setVisible(true)}
+            onClick={() => {
+              dispatch(commentActions.actions.setVisible(true));
```

Comment:
Same about the destructuring. You can destructure the setVisible method from the commentActions.actions

Reply:
Good idea, thanks!

Reply:
This options doesn't work, because I need to dispatch my action for change.